src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -229,27 +228,27 @@ EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
       event_engine_resolver_(std::move(event_engine_resolver)) {
   // Locking to prevent completion before all records are queried
   MutexLock lock(&on_resolved_mu_);
+  Ref(DEBUG_LOCATION, "OnHostnameResolved").release();
   hostname_handle_ = event_engine_resolver_->LookupHostname(
-      absl::bind_front(&EventEngineDNSRequestWrapper::OnHostnameResolved,
-                       Ref(DEBUG_LOCATION, "OnHostnameResolved")),
+      absl::bind_front(&EventEngineDNSRequestWrapper::OnHostnameResolved, this),
```

Comment:
I'd prefer to let the compiler enforce the ref ownership here.  How about something like this:

```
[self = Ref(DEBUG_LOCATION, "OnHostnameResolved")](
    absl::StatusOr<std::vector<EventEngine::ResolvedAddress>> addresses) {
  self->OnHostnameResolved(std::move(addresses));
}
```

Reply:
Done, here and elsewhere.